dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -943,6 +943,15 @@
 
 	<Target Name="_ComputeFrameworkAssemblies" DependsOnTargets="_ComputeVariables;ComputeResolvedFilesToPublishList">
 		<ItemGroup>
+			<!--
+				Remove runtime .dylib files for NativeAOT publishing. On osx-x64 and osx-arm64 we don't have
+				NativeAOT specific runtime packs yet (https://github.com/dotnet/runtime/issues/87060) so we
+				have to exclude several files from the standard runtime pack. If we didn't do that we would
+				end up linking with libclrgc.dylib and getting second copy of garbage collector, for example.
+			-->
+			<_XamarinFilesToDrop Include="@(ResolvedFileToPublish)" Condition="'$(_UseNativeAot)' == 'true' and '$(PublishAotUsingRuntimePack)' != 'true' and '%(ResolvedFileToPublish.DropFromSingleFile)' == 'true' and !$([System.String]::new('%(ResolvedFileToPublish.Filename)').StartsWith('libSystem'))"/>
+			<ResolvedFileToPublish Remove="@(_XamarinFilesToDrop)" Condition="'$(_UseNativeAot)' == 'true' and '$(PublishAotUsingRuntimePack)' != 'true'" />
+
```

Comment:
The `_ComputePublishLocation` seems like a better location for this, since that's where we do most of the other adjustments to `ResolvedFileToPublish`:

https://github.com/xamarin/xamarin-macios/blob/5bce8e71ab64513bc92f2a8861ad93eb066c5e6c/dotnet/targets/Xamarin.Shared.Sdk.targets#L1691-L1801

Reply:
This will be removed after the next SDK bump where we get osx-x64/osx-arm64 NativeAOT runtime packs.